apis/refs/v1beta1/computerefs.go

```diff
@@ -120,6 +120,102 @@ type ComputeAddressRef struct {
 	Namespace string `json:"namespace,omitempty"`
 }
 
+type ComputeAddress struct {
+	Project          string
+	Location         string
+	ComputeAddressID string
+	Address          string
+}
+
+func (c *ComputeAddress) Url() string {
+	if c.Location == "global" {
+		return fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/global/addresses/%s", c.Project, c.ComputeAddressID)
+	}
+	return fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/regions/%s/addresses/%s", c.Project, c.Location, c.ComputeAddressID)
+}
+
+func (c *ComputeAddress) GetAddress() string {
+	return fmt.Sprintf("%s", c.Address)
+}
+
+func ResolveComputeAddress(ctx context.Context, reader client.Reader, src client.Object, ref *ComputeAddressRef) (*ComputeAddress, error) {
+	if ref == nil {
+		return nil, nil
+	}
+
+	if ref.External != "" {
+		if ref.Name != "" {
+			return nil, fmt.Errorf("cannot specify both name and external on ComputeAddress reference")
+		}
+
+		tokens := strings.Split(ref.External, "/")
+		if len(tokens) == 5 && tokens[0] == "projects" && tokens[2] == "global" && tokens[3] == "addresses" {
+			return &ComputeAddress{
+				Project:          tokens[1],
+				Location:         "global",
+				ComputeAddressID: tokens[4]}, nil
+		} else if len(tokens) == 6 && tokens[0] == "projects" && tokens[2] == "location" && tokens[4] == "addresses" {
+			return &ComputeAddress{
+				Project:          tokens[1],
+				Location:         tokens[3],
+				ComputeAddressID: tokens[5]}, nil
```

Comment:
Address does not appear to be set in this code path

Reply:
For ComputeForwardingRule, the addressRef.External field accepts "address" of ComputeAddress([screenshot](https://screenshot.googleplex.com/C2sRUPo9BoTLTBY.png), KCC [doc](https://cloud.google.com/config-connector/docs/reference/resource-docs/compute/computeforwardingrule)). While other resources might accept the ID of ComputeAddress as we checked in the if-else statement. 

I need to add another if statement here, if ref.External is an ip address(i.e. 8.8.8.8):
```
// When External is an ip address
if len(tokens) == 1 {
  return &ComputeAddress{
  Address: tokens[0]}, nil
// When External is ComputeAddress ID
} else if len(tokens) == 5 && tokens[0] == "projects" && tokens[2] == "global" && tokens[3] == "addresses" {
...
```